src/math.rs

```diff
@@ -297,27 +347,12 @@ impl OrientedBeam {
         }
     }
 
-    pub fn intersects(&self, aabb: &Aabb3<f64>) -> bool {
-        intersects(&self.corners, &self.separating_axes, aabb)
-    }
-
-    pub fn contains(&self, p: &Point3<f64>) -> bool {
-        // What is the point in beam coordinates?
-        let Point3 { x, y, .. } =
-            self.isometry_inv.rotation.rotate_point(*p) + self.isometry_inv.translation;
-        x.abs() <= self.half_extent.x && y.abs() <= self.half_extent.y
-    }
-
-    pub fn transform(&self, isometry: &Isometry3<f64>) -> Self {
-        Self::new(isometry * &self.isometry_inv.inverse(), self.half_extent)
-    }
-
-    fn precompute_corners(
-        isometry: &Isometry3<f64>,
-        half_extent: &Vector2<f64>,
-    ) -> [Point3<f64>; 4] {
-        let corner_from = |x: f64, y: f64| {
-            isometry.rotation.rotate_point(Point3::new(x, y, 0.0)) + isometry.translation
+    fn precompute_corners(isometry: &Isometry3<S>, half_extent: &Vector2<S>) -> [Point3<S>; 4] {
+        let corner_from = |x, y| {
+            isometry
+                .rotation
+                .rotate_point(Point3::new(x, y, S::from(0.0).unwrap()))
```

Comment:
Nit: Use `S::zero()` instead of `S:from(0.0)`

Reply:
done